### PR TITLE
Add Lists+tables design guideline doc

### PR DESCRIPTION
**Pull request by mcarrano**
This PR adds the initial design guidelines documentation for Lists and Tables to the website.

Closes: https://github.com/patternfly/patternfly-next/issues/390

### CODE REVIEW

**Comment by mcarrano** [comment]
@stacymcauliffe I just added Lists and Tables. Please review when you have a chance. I already noticed that the jump links at the start of the doc need to be fixed, but I'll wait to see if you have other feedback before making more changes.

File: packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+![expandable data list](img/expandable-list.png)
+
+.
+1. **Expansion caret:** Toggles the expansion open and closed.
+2. **Expansion panel:** Revealed when the expansion is open. This is a container that can accept any supported layout to present some additional content to the user.
+
+#### When to use
+Use when you have more information than will comfortably fit inside a row or you want to provide a way for advanced users to access information that is not applicable to all users.
+
+## Data tables
```
Comment (stacymcauliffe) [inline packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md]: Is there a difference between what you refer to as styling here and what we talk about as a variation elsewhere?

File: packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+4. **Select checkbox:** Select this row.
+5. **Global actions:** These actions apply to all selected items.
+6. **Inline actions:** These actions apply only to the current row/item.
+7. **Pagination footer:** If present, navigation to a new page.
+
+#### When to use
+Use a Data Table when the information you want to display is fits into a structured, tabular format, i.e. has distinct rows and columns.
+
+#### When not to use
+The Data Table is not recommended for less structured of variably structured data that cannot be easily organized into columns. In these situations a Data Table should be used.
```
Comment (stacymcauliffe) [inline packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md]: Change "of" to "or"

File: packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+When to use
+Use a Compact Data Table when you want to show as much data per page as possible and readability is a secondary concern.
+
+
+### Expandable Data Table
+
+![expandable data table](img/expandable-data-table.png)
+
+1. **Expansion:** When present, expands the row.
```
Comment (stacymcauliffe) [inline packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md]: Try to keep a parallel structure to content in lists - for example...

1. Expansion: expands the row
2. Expansion panel: contains details associated with the row

1. Expandable cell: enables the user to expand any cell to reveal more information about an item. If a cell is already expanded, clicking on a different cell will close the current cell and open the new one.
2. Expansion panel: contains details associated with an expandable item

File: packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+![compound expandable data table](img/compound-expandable-data-table.png)
+
+1. **Expandable cell:** The user can click on any expandable cell to reveal more detail about an item. If the expansion for an item is already open, clicking on a different cell will close the current item and open a new one.
+2. **Expansion panel:** contains details associated with an expandable item.
+
+#### When to use
+Use a compound expandable list when you want multiple expansion panels that relate to specific table columns where it would not make sense to combine all of this information into a single, simple expansion.
+
+## Components and demos used
```
Comment (stacymcauliffe) [inline packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md]: I am actually digging having this content at the bottom of the guideline. Maybe we should move this globally. LMK what you think!

File: packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
+# Lists and tables
+PatternFly offers two components for displaying large data sets, the Data List and the Data Table. They satisfy similar use cases, but choosing the correct component to use in your design will be dependent on the type of data you need to display.
+
+Use [Data Lists](#Data lists) when:
```
[the remark this answers is not in the document]
Comment (mcarrano) [reply packages/patternfly-4/src/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md]: I don't feel strongly. Can change.

**Comment by mcarrano** [comment]
@stacymcauliffe I've made changes based on your feedback. I did not address the capitalization at this time as there were a bunch of places where I used caps for the component name. I think this is something we can clarify and clean-up later.

@stacymcauliffe @rachael-phillips please merge if this looks good enough for initial designer training.

**Pull request merged**